src/servers/src/prometheus.rs

```diff
@@ -603,6 +615,27 @@ pub async fn labels_query(
     PrometheusJsonResponse::success(PrometheusResponse::Labels(sorted_labels))
 }
 
+async fn get_all_column_names(
+    catalog: &str,
+    schema: &str,
+    manager: &CatalogManagerRef,
+) -> std::result::Result<Vec<String>, catalog::error::Error> {
+    let table_names = manager.table_names(catalog, schema).await?;
+
+    let mut labels = HashSet::new();
+    for table_name in table_names {
+        let Some(table) = manager.table(catalog, schema, &table_name).await? else { continue };
+        let schema = table.schema();
+        for column in schema.column_schemas().iter() {
```

Comment:
style(non-blocking): `column_schemas()` returns a slice
```suggestion
        for column in schema.column_schemas() {
```